katharsis-core/src/main/java/io/katharsis/queryspec/DefaultQuerySpecDeserializer.java

```diff
@@ -240,9 +240,13 @@ private void deserializeFilter(QuerySpec querySpec, Parameter parameter) {
 		Class<?> attributeType = getAttributeType(querySpec, attributePath);
 		Set<Object> typedValues = new HashSet<>();
 		for (String stringValue : parameter.values) {
-			@SuppressWarnings({ "unchecked", "rawtypes" })
-			Object value = typeParser.parse(stringValue, (Class) attributeType);
-			typedValues.add(value);
+			if (stringValue.toLowerCase().equals("null")) {
+				typedValues.add(null);
+			} else {
+				@SuppressWarnings({ "unchecked", "rawtypes" })
+				Object value = typeParser.parse(stringValue, (Class) attributeType);
+				typedValues.add(value);
```

Comment:
I am sorry but I don't agree with you because if the parameter value is equals to "null" it should create a null object instead if I moved outside the if/else your 3 lines  the value will be overwrite with the "null" string.